src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor({ health = 100, name }) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(bittedAnimal) {
+    if (bittedAnimal instanceof Herbivore && !bittedAnimal.hidden) {
+      bittedAnimal.health -= 50;
+    }
+
+    if (bittedAnimal.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== bittedAnimal);
```

Comment:
The check for the animal's health should be inside the previous 'if' block to ensure that it only applies to herbivores that were actually bitten and not hidden. As it stands, the health check could incorrectly filter out any carnivore or hidden herbivore that has health less than or equal to 0.